tests/utils/runner/runner.go

```diff
@@ -0,0 +1,377 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package runner
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	runner_sdk "github.com/ava-labs/avalanche-network-runner/client"
+	"github.com/ava-labs/avalanche-network-runner/rpcpb"
+	runner_server "github.com/ava-labs/avalanche-network-runner/server"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/subnet-evm/plugin/evm"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+// Subnet provides the basic details of a created subnet
+// Note: currently assumes one blockchain per subnet
+type Subnet struct {
+	// SubnetID is the txID of the transaction that created the subnet
+	SubnetID ids.ID
+	// Current ANR assumes one blockchain per subnet, so we have a single blockchainID here
+	BlockchainID ids.ID
+	// ValidatorURIs is the base URIs for each participant of the Subnet
+	ValidatorURIs []string
+}
+
+type ANRConfig struct {
+	LogLevel            string
+	AvalancheGoExecPath string
+	PluginDir           string
+	GlobalNodeConfig    string
+}
+
+// NetworkManager is a wrapper around the ANR to simplify the setup and teardown code
+// of tests that rely on the ANR.
+type NetworkManager struct {
+	ANRConfig ANRConfig
+
+	subnets []*Subnet
+
+	logFactory      logging.Factory
+	anrClient       runner_sdk.Client
+	anrServer       runner_server.Server
+	done            chan struct{}
+	serverCtxCancel context.CancelFunc
+}
+
+func NewDefaultANRConfig() ANRConfig {
+	defaultConfig := ANRConfig{
+		LogLevel:            "info",
+		AvalancheGoExecPath: os.ExpandEnv("$GOPATH/src/github.com/ava-labs/avalanchego/build/avalanchego"),
+		PluginDir:           os.ExpandEnv("$GOPATH/src/github.com/ava-labs/avalanchego/build/plugins"),
+		GlobalNodeConfig: `{
+			"log-display-level":"info",
+			"proposervm-use-current-height":true,
+			"throttler-inbound-validator-alloc-size":"107374182",
+			"throttler-inbound-node-max-processing-msgs":"100000",
+			"throttler-inbound-bandwidth-refill-rate":"1073741824",
+			"throttler-inbound-bandwidth-max-burst-size":"1073741824",
+			"throttler-inbound-cpu-validator-alloc":"100000",
+			"throttler-inbound-disk-validator-alloc":"10737418240000",
+			"throttler-outbound-validator-alloc-size":"107374182"
+		}`,
+	}
+	// If AVALANCHEGO_BUILD_PATH is populated, override location set by GOPATH
+	if envBuildPath, exists := os.LookupEnv("AVALANCHEGO_BUILD_PATH"); exists {
+		defaultConfig.AvalancheGoExecPath = fmt.Sprintf("%s/avalanchego", envBuildPath)
+		defaultConfig.PluginDir = fmt.Sprintf("%s/plugins", envBuildPath)
+	}
+	return defaultConfig
+}
```

Comment:
can we make this a var?

Reply:
It currently uses this function so that it can expand environment variables when it's called. We could do the same in `init()`, but since we use environment variables here, I'd prefer to expand them when the function is called.